### ISSUE TITLE
Add Assertion that carriers exist within eligible sets

### ISSUE THREAD

**Issue by ktehranchi** [PyPSA/pypsa-usa#382]
### Checklist

- [X] I am using the current [`master`](https://github.com/PyPSA/pypsa-usa/tree/master) branch
- [X] I am running on an up-to-date [`pypsa-usa` environment](https://github.com/PyPSA/pypsa-usa/blob/master/workflow/envs/environment.yaml). Update via `conda env update -f envs/environment.yaml`

### The Issue

If a user mistakenly places a conventional carrier in StorageUnits, instead of skipping it, we should set a warning that the carrier is not eligible. 

Ideally, this happens early in each script, or in snakemake, so that the user gets the feedback immediately

**Comment by trevorb1** [comment]
I agree! Once we get to more stable configuration file(s) setup, it may be good to write some simple logic to validate the configuration file. [Pydantic](https://docs.pydantic.dev/latest/) could be a good option (easy to setup and can handle conditional logic quite nicely)! That way we can just run the validation script once at the top of the workflow (maybe via the [onstart](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#onstart-onsuccess-and-onerror-handlers) command) so none of the checking has to happen in the scripts. :)
